cime_config/acme/machines/Makefile

```diff
@@ -450,8 +491,13 @@ endif
 
 # System libraries (netcdf, mpi, pnetcdf, esmf, trilinos, etc.)
 ifndef SLIBS
-   SLIBS := -L$(LIB_NETCDF) -lnetcdf
+  ifeq ($(NETCDF_SEPARATE), false)
+    SLIBS := -L$(LIB_NETCDF) -lnetcdf
+  else ifeq ($(NETCDF_SEPARATE), true)
+    SLIBS := -L$(LIB_NETCDF_C) -L$(LIB_NETCDF_FORTRAN) -lnetcdf
```

Comment:
Actually you are only linking the netcdf c library here - the netcdf fortran library is netcdff.
You should be able to modify these SLIBS lines to have both, and I believe that it will then work for both shared and static links
-lnetcdf -lnetcdff